specs-go/config.go

```diff
@@ -16,8 +16,9 @@ type Spec struct {
 
 // Device is a "Device" a container runtime can add to a container
 type Device struct {
-	Name           string         `json:"name"`
-	ContainerEdits ContainerEdits `json:"containerEdits"`
+	Name           string            `json:"name"`
+	Annotations    map[string]string `json:"annotations,omitempty"`
```

Comment:
@klihub I added them "there" because they are not used for any edits of the container. Meta information about the device. Those are annotations per device; if we would need annotations per spec they would need to move to:

```diff
type Spec struct {
	Version string `json:"cdiVersion"`
	Kind    string `json:"kind"`
+       Annotations map[string]string `json:"annotations,omitempty"`
	Devices        []Device       `json:"devices"`
	ContainerEdits ContainerEdits `json:"containerEdits,omitempty"`
}
```
that's why I added them per device: 

```diff
type Device struct {
	Name           string            `json:"name"`
+	Annotations    map[string]string `json:"annotations,omitempty"` // Annotations add meta information per device
	ContainerEdits ContainerEdits    `json:"containerEdits"`
}

```

The annotations here are needed for the runtime not the actual container. If we would need to add annotations to the container, they would need to go to the `ContainerEdits` struct.

Reply:
Ok, thanks. I missed that it's for consumption by the runtime. Now it makes sense.